### PR TITLE
Adding package promotion scripts (squashed)

### PR DESCRIPTION
**Pull request by jonahjon**
Signed-off-by: jonahjon <jonahjones094@gmail.com>

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by jonahjon** [comment]
O yay, all the tests passed!

File: build/lib/helm_promotion.sh [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ECR_REPO=$(echo ${PROJECT_PATH} | sed "s/aws\///") # We sed to remove aws/ from ECR repo names
+echo "ECR_REPO=${ECR_REPO}"
+GIT_TAG=$(cat ${BASE_DIRECTORY}/projects/${PROJECT_PATH}/GIT_TAG)
+
+export IMAGE_TAG=${GIT_TAG}-${GIT_HASH}
+echo "IMAGE_TAG=${IMAGE_TAG}"
+
+# Pull Helm chart from private ECR
+aws ecr get-login-password --region us-west-2 | HELM_EXPERIMENTAL_OCI=1 helm registry login --username AWS --password-stdin ${ACCOUNT_ID}.dkr.ecr.us-west-2.amazonaws.com
+helm pull oci://${ACCOUNT_ID}.dkr.ecr.us-west-2.amazonaws.com/${ECR_REPO} --version ${IMAGE_TAG}-helm
```
Comment (vincentni) [inline build/lib/helm_promotion.sh:38]: Is it better that we pass AWS_REGION from Makefile to here instead of hardcoding it to us-west-2?

**Review by vincentni — APPROVED** [review]

File: build/lib/helm_promotion.sh (hunk 1 above)
Comment (jonahjon) [reply build/lib/helm_promotion.sh:38]: At the moment we're hardcoding this in the artifact bucket scripts, so it's not super needed unless we change everything else such as.

https://github.com/aws/eks-anywhere-build-tooling/blob/main/config/docker-ecr-config.json#L6
https://github.com/aws/eks-anywhere-build-tooling/blob/main/build/lib/common.sh#L284

File: Makefile [hunk 2]
```diff
@@ -1,7 +1,9 @@
 BASE_DIRECTORY=$(shell git rev-parse --show-toplevel)
+BUILD_LIB=${BASE_DIRECTORY}/build/lib
 AWS_ACCOUNT_ID?=$(shell aws sts get-caller-identity --query Account --output text)
 AWS_REGION?=us-west-2
 IMAGE_REPO?=$(if $(AWS_ACCOUNT_ID),$(AWS_ACCOUNT_ID).dkr.ecr.$(AWS_REGION).amazonaws.com,localhost:5000)
+ECR_PUBLIC_URI?=$(shell AWS_REGION=us-east-1 && aws ecr-public describe-registries --query 'registries[*].registryUri' --output text)
```
[the remark this answers is not in the document]
Comment (jonahjon) [reply Makefile:6]: will change this

File: Makefile [hunk 3]
```diff
@@ -1,7 +1,9 @@
 BASE_DIRECTORY=$(shell git rev-parse --show-toplevel)
+BUILD_LIB=${BASE_DIRECTORY}/build/lib
 AWS_ACCOUNT_ID?=$(shell aws sts get-caller-identity --query Account --output text)
 AWS_REGION?=us-west-2
 IMAGE_REPO?=$(if $(AWS_ACCOUNT_ID),$(AWS_ACCOUNT_ID).dkr.ecr.$(AWS_REGION).amazonaws.com,localhost:5000)
+ECR_PUBLIC_URI?=$(shell aws ecr-public describe-registries --region us-east-1 --query 'registries[*].registryUri' --output text)
```
Comment (lewisdiamond) [inline Makefile:6]: Wouldn't this return a list of registries if more than one exist?

Comment (jonahjon) [reply Makefile:6]: yes, but I think for the time being you are only allowed 1 public registry per account.

We could change it to be `registries[0]` if you prefer though.

File: Makefile [hunk 4]
```diff
@@ -102,6 +105,9 @@ stop-buildkit-and-registry:
 generate:
 	build/lib/generate_projects_list.sh $(BASE_DIRECTORY)
 
+helm/promotion:
```
[the remark this answers is not in the document]
Comment (jonahjon) [reply Makefile:108]: fixing that.

**Pull request merged**